pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/HashUtils.java

```diff
@@ -36,6 +40,31 @@ public static byte[] hashMD5(byte[] bytes) {
     return Hashing.md5().hashBytes(bytes).asBytes();
   }
 
+  /**
+   * Returns a byte array that is a concatenation of the binary representation of each of the passed UUID values. This
+   * is done by getting a String from each value by calling {@link Object#toString()}, which is then used to create a
+   * {@link UUID} object. The 16 bytes of each UUID are appended to a buffer which is then returned in the result.
+   * If any of the values is not a valid UUID, then we return the result of {@link #concatenate}.
+   */
+  public static byte[] hashUUID(Object[] values) {
+    byte[] result = new byte[values.length * 16];
+    ByteBuffer byteBuffer = ByteBuffer.wrap(result).order(ByteOrder.BIG_ENDIAN);
+    for (Object value : values) {
+      if (value == null) {
+        return concatenate(values);
+      }
+      UUID uuid;
+      try {
+        uuid = UUID.fromString(value.toString());
+      } catch (Throwable t) {
```

Comment:
Since this is consider to be rare, can we log the primary value here?

Reply:
This is to be used in the ingestion flow. We can log every few batch of events. I didn't want to do it as part of this PR since the ideal solution would likely require Thread locals and it might become complex to figure out.